### PR TITLE
JENKINS-3549: display promotion stars on the view.

### PR DESCRIPTION
**Pull request by dnozay**
alternative to pull request #30. see also https://issues.jenkins-ci.org/browse/JENKINS-3549

### CODE REVIEW

File: src/main/java/hudson/plugins/promoted_builds/Status.java [hunk 1]
```diff
@@ -271,6 +271,12 @@ public Promotion getLast() {
         return null;
     }
 
+    public Boolean isLastAnError() {
```
Comment (daniel-beck) [inline src/main/java/hudson/plugins/promoted_builds/Status.java]: Would it be useful to make new public methods that only exist to serve the UI `@Restricted(NoExternalUse.class)`?

File: src/main/resources/hudson/plugins/promoted_builds/PromotionStatusColumn/column.jelly [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:l="/lib/layout">
+  <td>
+    <j:set var="ppa" value="${it.getAction(job)}"/>
+    <j:choose>
+      <j:when test="${ppa!=null}">
+        <j:set var="pp" value="${ppa.getPromotionProcesses()}"/>
+        <j:choose>
+          <j:when test="${pp!=null}">
```
Comment (daniel-beck) [inline src/main/resources/hudson/plugins/promoted_builds/PromotionStatusColumn/column.jelly]: `getPromotionProcesses()` does not return `null`, just an empty list. Have you tested this?

Comment (dnozay) [inline src/main/resources/hudson/plugins/promoted_builds/PromotionStatusColumn/column.jelly]: I had forgotten that I made it return an empty list so I would need a `<j:choose>` / `<j:when>`... It is possible to manually check all other cases by configuring a view with the column and some sample jobs. I can simplify this jelly. Is that ok?

File: src/main/java/hudson/plugins/promoted_builds/Status.java (hunk 1 above)
Comment (dnozay) [inline src/main/java/hudson/plugins/promoted_builds/Status.java]: sure, I can try that.

**Comment by oleg-nenashev** [comment]
Looks good to me :+1: , pom changes may conflict with #61

**Comment by oleg-nenashev** [comment]
Reopening it to trigger the PR builder & static analysis

**Pull request merged**